__tests__/dependencies.ts

```diff
@@ -96,11 +96,21 @@ describe("Manually constructed precedence graphs", () => {
     expect(graph.size()[0]).toEqual(6); // V
     expect(graph.size()[1]).toEqual(7); // E
     expect(graph.toString()).toBe(
-      StringUtil.dontIndent`digraph G {
-            "app.R[R0]"->"app.R[R1]"->"app.R[R4]"->"app.R[R3]"->"app.R[R5]";
-            "app.R[R0]"->"app.R[R4]";
-            "app.R[R1]"->"app.R[R2]"->"app.R[R3]";
-            }`
+      StringUtil.dontIndent
```

Comment:
These tests might need to be changed, as the representation of `mermaid.js` here does not have to conform to a specific ordering.

Reply:
What do you mean, exactly? Is the function that produces the output printing things in an arbitrary order?

Reply:
It is not printing things in arbitrary order. Nevertheless, our test should not expect it to be printing things in some certain order

Reply:
Why not?

Reply:
It relies on 1. the fact that JS sets maintain insertion order, and 2. we are looping in that order. If there is an implementation that does not loop in such an order, or if 1 does not hold for `nodejs` anymore (which is unlikely), then the tests will break while in reality it shouldn't. I don't think that this is worth putting too much effort on this as of now, but we (probably) should bear this in mind.

Reply:
We are testing our own implementation, and if it breaks because of changes in the underlying data structures (which seems unlikely to happen), then this leading to test failures would be fine.